python/pyarrow/array.pxi

```diff
@@ -997,14 +997,17 @@ cdef class Array(_PandasConvertible):
 
         return wrap_datum(out)
 
-    def filter(self, Array mask):
+    def filter(self, Array mask, drop_nulls=True):
         """
         Filter the array with a boolean mask.
 
         Parameters
         ----------
         mask : Array
             The boolean mask indicating which values to extract.
+        drop_nulls : bool, default True
```

Comment:
If we might want to add other options in the future (there were some others raised in the issue I think?), a boolean might not be the best value.

Reply:
Looking back at ARROW-8235, the other one mentioned was "KEEP". I don't know if we have plans to add that in the future? Maybe not in the short term? 
(and another one could be "ERROR")

Reply:
I'll change this to a string parameter to ease addition of other options in a follow up.